.github/workflows/azure-deploy.yml

```diff
@@ -0,0 +1,138 @@
+        uses: actions/checkout@v4
+
+      - name: Setup Node ${{ env.NODE_VERSION }} Environment
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - uses: oven-sh/setup-bun@v2
+
+      - name: 'Resolve Project Dependencies Using Bun'
+        shell: pwsh
+        run: |
+          pushd './${{ env.AZURE_FUNCTIONAPP_PACKAGE_PATH }}'
+          bun install --frozen-lockfile
+          bun run jest:test
+          bun run build
+          bun rimraf node_modules
+          bun install --frozen-lockfile --production
+          popd
+
+      - name: Upload artifact for deployment job
+        uses: actions/upload-artifact@v4
+        with:
+          name: node-app
+          path: .
+
+  update-environment:
+    runs-on: ubuntu-latest
+    needs: build
+
+    permissions:
+      id-token: write
+
+    steps:
+      - name: Generate local.settings.json
+        env:
+          GITHUB_SECRETS: ${{ toJson(secrets) }}
+        shell: bash
+        run: |
+          # Parse the JSON secrets
+          secrets=$(echo "$GITHUB_SECRETS" | jq -c '.')
+
+          # Create the base settings object
+          settings=$(jq -n '{
+            "IsEncrypted": false,
+            "Values": {}
+          }')
+
+          # Define excluded prefixes
+          excluded_prefixes=("GITHUB_" "ACTIONS_" "AZUREAPPSERVICE_")
+
+          # Iterate through secrets and add non-excluded ones
+          for secret_name in $(echo "$secrets" | jq -r 'keys[]'); do
+            # Check if the secret should be included
+            include=true
+            for prefix in "${excluded_prefixes[@]}"; do
+              if [[ "$secret_name" == "$prefix"* ]]; then
+                include=false
+                break
+              fi
+            done
+
+            # Add the secret if it passes the filter
+            if [ "$include" = true ]; then
+              secret_value=$(echo "$secrets" | jq -r ".[\"$secret_name\"]")
+              settings=$(echo "$settings" | jq --arg name "$secret_name" --arg value "$secret_value" '.Values[$name] = $value')
+            fi
+          done
+
+          # Output the final settings to local.settings.json
+          echo "$settings"
+          echo "$settings" | jq '.' > local.settings.json
+
+      - uses: azure/login@v2
+        with:
+          client-id: ${{ secrets.AZUREAPPSERVICE_CLIENTID }}
+          tenant-id: ${{ secrets.AZUREAPPSERVICE_TENANTID }}
+          subscription-id: ${{ secrets.AZUREAPPSERVICE_SUBSCRIPTIONID }}
+
+      - name: Azure CLI script to upload environment
+        uses: azure/cli@v2
+        with:
+          azcliversion: latest
+          inlineScript: |
+            az account show
+            az functionapp config appsettings set \
+              --name ubiquity-os \
+              --resource-group ubiquity-os_group \
+              --settings @local.settings.json
+
+  deploy:
+    runs-on: windows-latest
+    needs: build
```

Comment:
shouldn't this be run after `update-environment`? I was recently using Azure App Service and when I changed env I had to restart the app

Reply:
I think both steps restart the app, so it doesn't matter much. The best way would be to combine the two steps, the command line can handle upload + local.storage.settings upload but the GitHub Action does not support it.

Reply:
> GitHub Action does not support it.

That seems unlikely because its a fully capable ubuntu box.